src/qa4sm_reader/plotting_methods.py

```diff
@@ -1509,24 +1602,61 @@ def _extent2dpi_fraction(extent):
     # converts a certain validation extent to a 0-1 fraction
     # indicating the output quality
     max_extent = 360 * 110
-    actual_extent = (extent[1]-extent[0]) * (extent[3]-extent[2])
+    actual_extent = (extent[1] - extent[0]) * (extent[3] - extent[2])
 
     return actual_extent / max_extent
 
 
-def output_dpi(
-        res,
-        units,
-        extent,
-        dpi_min=globals.dpi_min,
-        dpi_max=globals.dpi_max
-) -> float:
+def output_dpi(res,
+               units,
+               extent,
+               dpi_min=globals.dpi_min,
+               dpi_max=globals.dpi_max) -> float:
     # get ouput dpi based on image extent and validation resolution
     # dpi = SQRT(extent_coeff^2 + res_coeff^2)
-    dpi_vec = _extent2dpi_fraction(extent)**2 + _res2dpi_fraction(res, units)**2
+    dpi_vec = _extent2dpi_fraction(extent)**2 + _res2dpi_fraction(res,
+                                                                  units)**2
     dpi_vec = np.sqrt(dpi_vec)
     dpi_fraction = dpi_vec / np.sqrt(2)
 
-    dpi = dpi_min + (dpi_max-dpi_min) * dpi_fraction
+    dpi = dpi_min + (dpi_max - dpi_min) * dpi_fraction
 
     return float(dpi)
+
+
+def average_non_additive(values: pd.Series, nobs: pd.Series) -> float:
```

Comment:
I would use https://numpy.org/doc/stable/reference/generated/numpy.arctanh.html for the z scores, but I guess it doesn't make a difference (maybe the atan<sup>⁻1</sup> is better when r<sub>i</sub> are ~1?).
I would only support numpy arrays from the start, make sure that their lengths match, or if you want to allow arrays and Series, make sure to use `typing.Union[pd.Series, np.ndarray]` in the type hints.
Maybe use something like `mask = np.isfinite(values) & np.isfinite(nobs)` `nobs = nobs[mask]` `values=values[mask]` instead of argwhere and delete (probably faster and safer).

Shouldnt this `mean = np.sum((nobs - 3) * z_scores) / (np.sum(nobs) - 3 * k)` be
` mean = np.sum((nobs - 3) * z_scores) / np.sum(nobs - (3 * k))`